src/main/java/uk/ac/cam/cl/dtg/segue/api/GroupsFacade.java

```diff
@@ -400,9 +400,22 @@ public Response editGroup(@Context final HttpServletRequest request, final UserG
                 return new SegueErrorResponse(Status.NOT_FOUND, "Group specified does not exist.").toResponse();
             }
 
+            // If user is not the owner...
             if (!existingGroup.getOwnerId().equals(user.getId())) {
-                return new SegueErrorResponse(Status.FORBIDDEN,
-                        "The group you have attempted to edit does not belong to you.").toResponse();
+                // Check if an additional manager with privileges is allowed to make these changes
+                boolean ownerOnlyFieldChanged = !existingGroup.getOwnerId().equals(groupDTO.getOwnerId())
```

Comment:
This is kind of nasty, but I can't see a reason why it wouldn't work. 
I think that in edge cases where this returns `true` when it probably shouldn't, something weird has happened in the front-end to mutate the group object.

Reply:
Looking into the DTO pattern [^1][^2] a bit, they exist to hold only the fields necessary for a particular interaction - that sounds relevant here as there are fields we should not be able to update from this endpoint at all, and some that depend on the user/group relationship. Perhaps creating new DTOs reflecting only the fields that can be updated in each situation would be cleaner, along with versions of `groupManager::editUserGroup()` to process them. That would be a bigger change of course!

[^1]: https://auth0.com/blog/automatically-mapping-dto-to-entity-on-spring-boot-apis/#What-is-a-DTO
[^2]: https://snyk.io/blog/how-to-use-java-dtos/

Reply:
I fully agree, but it would be a bigger change. The immense amount of boilerplate required for simple things like this is annoying to say the least. Since (I think) we want admins to be able to edit group objects as they please, we would need a different endpoint that accepts the smaller DTO, and as you say new versions of `editUserGroup` etc. 
Considering there are quite a few endpoints where we make this mistake already, I'd rather make a card for targeting all of them in one clean refactor?